packages/cozy-harvest-lib/src/policies/clisk.js

```diff
@@ -53,7 +94,21 @@ function onLaunch({ konnector, account, trigger }) {
         }
       })
     )
-  }
+  })
+}
+
+/**
+ * Listens to postMessage event. Compatible with android and ios webviews
+ *
+ * @param {Function} callback - callback function
+ */
+function addMessageListener(callback) {
+  window.document?.addEventListener?.('message', function (e) {
+    callback(e.data)
+  })
+  window?.addEventListener?.('message', function (e) {
+    callback(e.data)
+  })
 }
```

Comment:
not blocking for now, but can't we start to use cozy-intent instead of calling postMessage / listener manually? cc @Ldoppea

Reply:
We discussed about that yesterday.

If we want to use cozy-intent here, then we will have to rethink the flagship app architecture:
- current postMessage are handled by the HomeView (cf [this code](https://github.com/cozy/cozy-flagship-app/blob/master/src/screens/home/components/HomeView.js#L216-L217)) and then we set the Launcher state using a drilled prop
- if we use cozy-intent, then the message won't be intercepted by HomeView anymore but by the `localMethods` (cf [this code](https://github.com/cozy/cozy-flagship-app/blob/master/src/App.js#L177)). Then we should find a way to interact with the Launcher from `localMethods` (this can be done using the same LauncherEvent item we implemented in [this PR](https://github.com/cozy/cozy-flagship-app/pull/659))

So if we want to use cozy-intent we may want to rework this part, and we may want to extract the LauncherView from the HomeScreen component (as  the drilled `setLauncherContext` prop is the only reason for having them in the same component tree)

One last thing, we heard that MesPapiers may need to launch konnectors in the future, if this is the case, there is one more reason to do this rework because today only the HomeView can do this. But we will also have to think about how we should handle app permissions for this.

Reply:
> So if we want to use cozy-intent we may want to rework this part, and we may want to extract the LauncherView from the HomeScreen component

please yep 🙏 
https://github.com/cozy/cozy-flagship-app/pull/534#discussion_r1066026603